Rewrite README

glasserc opened the pull request:
While trying to resolve #42, I had occasion to reread the README for this project and realized that a lot of it is abrasive chest-thumping which doesn't even do the main thing a README should do, which is explain what the package is, what it does, and how to use it. This is an attempt to tone it down a little bit and focus on what's important while trying to keep at least a little bit of the character (most of which is moved to a FAQ at the bottom).

You can see it rendered at https://github.com/glasserc/ethan-wspace/tree/rewrite-readme.

@rkachach, do you think this would have helped you if you had read it first?

glasserc committed:
Remove some of the ranting and try to make it clearer how to use it

glasserc committed:
Try to condense ranting

Move some rants into documentation. Smart Tabs is the topic of two of
these rants so try to reduce duplication a little.

glasserc committed:
Add a screenshot

rkachach commented:
Thanks for dedicating time to improving the Readme. Just a couple of suggestions (if you don't mind) around the Readme file:

1) I think the good thing about the ethan-mode is that it's non-intrusive and tries to do the best it can without interfering in the user changes. Thus, I think instead of  **an opinionated** I find non-intrusive a better description for the mode.

2) The yellow image under `The solution: ethan-wspace.` is very large and takes a lot of space moving more important content **How it works** bellow in the screen. I think it would be better to make this image smaller and move it after the **How it works** section. 

I think that would make the README shorter and clear.

glasserc commented:
Thanks for the feedback! I agree that "non-intrusive" is a better way to describe the philosophy of `ethan-wspace` so I will definitely make that change.

The yellow image is a screenshot of the mode "in action". (Maybe that isn't clear because of my emacs configuration. I could add something like "This is what it looks like in action:" if you think that would help.) I think it's pretty common for casual users to want to get a sense of how the software looks before deciding whether to investigate more effort into understanding it, and I don't think it takes a lot of effort to glance at the screenshot and decide whether it's interesting or not before scrolling down further (if necessary). For this reason I think it's important to have something before the "How it works" section (although I am open to the idea of it being a different screenshot if this one is confusing or something). For comparison, Magit does the same thing: https://magit.vc/.

glasserc committed:
Change "opinionated" to "non-intrusive" to better reflect philosophy

Thanks @rkachach for the suggestion.

glasserc committed:
Move screenshot to after the "How it works" section

This fits better since you can't really understand the screenshot
without understanding what it's doing.

Thanks @rkachach for the suggestion.

glasserc commented:
Thinking about it more, I think you're right about the position of the screenshot too. It's hard to understand what the screenshot is showing you without the background of what the software does. I've made that change too.

rkachach commented on README.rst:
```diff
@@ -1,73 +1,44 @@
+  highlighting is added.
 
-ethan-wspace takes the approach of *do no harm*. Specifically, when you open a file:
+It looks like this:
 
```
What about adding something like the following to this sentence?
Where **ew:mnLT**  represents the current space-status of the buffer -- and a description of what each letter means-- (otherwise the reader will just see the red spaces and doesn't notice the importance of these letters)

glasserc replied:
I think some of this is covered by the "Usage" section, which is quite low in the document. What if I move it up a bit? Do you think that covers enough of this information?

rkachach replied:
Do you mean this section?

`There's also a mode line "lighter" for ethan-wspace which looks something like ew:mnLt. Lower case letters indicate categories of whitespace which will be cleaned before save; upper case letters indicate categories which are being highlit. You can click on a particular letter to change from one to the other.`

I think maybe it's better to have it a little bit upper and explain what each letter means. Reading this paragraph it's hard to the user to figure out what they mean.

glasserc replied:
The problem is that with four letters, there's a lot that would need to be conveyed all at once. Currently this information is all in the "Errors" section. I've tried to make a clearer reference to that and try to explain the modeline lighter information sooner. What do you think?

rkachach replied:
Sorry for the late response! yeah, I think a simple explanation at the beginning of the Readme of what the letters mean could be helpful for the user. Anyway when the buffer is dirty you are already highlighting the bad spaces which is the most important visual information/feedback you can give to the user. The minor mode letters are a minor issue IMHO.

glasserc replied:
OK, I think this is enough of an improvement over what we have now that I'm going to merge it. If you have further suggestions for the text around the screenshot, please feel free to open a PR!

rkachach commented:
@glasserc it looks way better and easier to understand. Thank you very much for considering my suggestions :)

glasserc committed:
The "usage" section is important to understand the screenshot

So let's move it closer.

glasserc committed:
Rewrite usage section to try to order by priority

Let's start with the information that is most relevant to someone
looking at the screenshot, and then explain in greater detail as we
go.

Configuration information about the face can go in the "installation"
section, or some other section more focused on configuration.

Thanks @rkachach for the suggestion.

glasserc committed:
Add one more quick note about how categories are independent
